iwyu_tool_test.py

```diff
@@ -208,6 +259,40 @@ def test_consecutive_spaces(self):
         self.assert_win_split('clang  -I. \t     -A',
                               ['clang', '-I.', '-A'])
 
+class BootstrapTests(unittest.TestCase):
+    def setUp(self):
+        self.iwyu_tool_main_mock =  MockIwyuToolMain()
+        self.sys_exit_stub = StubSysExit()
+        self.sys_exit_stub.will_return(0)
+
+    def tearDown(self):
+        self.iwyu_tool_main_mock.reset()
+        self.sys_exit_stub.reset()
+
+    def test_argument_parser_sets_argument_correctly(self):
+        """ Check that arguments are verbatim injected to iwyu. """
+        argv_stub = StubSysArgv(["iwyu_tool.py", "-p", ".", "--", "arg1"])
+        iwyu_tool._bootstrap()
+        iwyu_call_args = self.iwyu_tool_main_mock.get_call_args()
+        self.assertIn(["arg1"], iwyu_call_args)
+
+    def test_argument_parser_does_include_xiwyu_flag(self):
+        """ Check that -Xiwyu and --iwyu_opt arguments are passed to the iwyu call. """
+        argv_stub = StubSysArgv(["iwyu_tool.py", "-p", ".", "--", "arg1", "-Xiwyu", "arg2", "--iwyu_opt", "arg3"])
+        iwyu_tool._bootstrap()
+        iwyu_call_args = self.iwyu_tool_main_mock.get_call_args()
+        self.assertIn(["arg1", "-Xiwyu", "arg2", "--iwyu_opt", "arg3"], iwyu_call_args)        
+
+    def test_argument_parser_uses_the_first_separator_for_splitting_arguments(self):
```

Comment:
Try shortening this method name so it fits within 80-column width

Reply:
Done ✔️